### PR TITLE
2355 edit remove constraints

### PR DESCRIPTION
**Pull request by gonzalezma**
## Description

Fixes the problems with the remove/edit constraint options, including the constraints on polydisperse parameters.

Note that the validation of the formula has been disabled, as it is not working properly and this will require much more thought to fix. A separate issue (#2740) has been created for this.

Fixes # 2355

## How Has This Been Tested?

Create a project, define a few constraints (including polydisperse parameters) and play removing and editing them from the right mouse button menu.

## Review Checklist (please remove items if they don't apply):

- [ ] Code has been reviewed
- [ ] Functionality has been tested
- [ ] Windows installer (GH artifact) has been tested (installed and worked) 
- [ ] MacOSX installer (GH artifact) has been tested (installed and worked) 
- [ ] User documentation is available and complete (if required)
- [ ] Developers documentation is available and complete (if required)
- [ ] The introduced changes comply with SasView license (BSD 3-Clause)

### CODE REVIEW

**Comment by wpotrzebowski** [comment]
I am not sure if I have encountered know issue but I am getting following error:
Screenshot 2024-01-18 at 22 06 25

**Comment by wpotrzebowski** [comment]
Error appeared when I tried to constrain polydispersity parameters
`22:05:57 - ERROR: Traceback (most recent call last): File "sas/qtgui/Perspectives/Fitting/ConstraintWidget.py", line 1129, in onAcceptConstraint File "sas/qtgui/Perspectives/Fitting/FittingWidget.py", line 961, in addConstraintToRow TypeError: '`

**Comment by gonzalezma** [comment]
You are right. I think that this could be because at some point I was working on another branch to address #2356. This could mean that this branch would not work with the PD parameters. I thought that I had tested for both types, though ... Perhaps I was on the wrong branch? I don't know which is the best way to proceed. If you want, I can create a new branch with the fixes for both #2355 and #2356 and create a single PR request from there, and we forget about this PR and the other one. What do you think?

**Comment by wpotrzebowski** [comment]
@gonzalezma I think we can merge both at the same time. We just need to make sure what to test.

**Comment by gonzalezma** [comment]
That sounds good. For info, a large part of the issues with the PD parameters is that they have a "coding name" (e.g. radius.width) and a "human name" (e.g. Distribution of radius), and depending on the workflow sometimes constraint.param has the coding name and others it has the human name, so it is needed to check and convert names before doing a comparison.
When I was testing this, I was using a saved project (and then adding the PD constraints manually, as those are not loaded when loading a project - issue #2735) and everything worked fine. But when I start from scratch I get the same error as you. I have some hopes that the fix for #2356 could solve this, so I would propose the following:
1. Revise this PR ignoring the PD constraints for now.
2. Revise the PR for #2356.
3. Once that both are accepted and merged with the release branch we can check if add/remove/edit constraints works also as expected for the PD ones.
Does this sound reasonable to you?

File: src/sas/qtgui/Perspectives/Fitting/FittingWidget.py [hunk 1]
```diff
@@ -1075,6 +1078,8 @@ def editConstraint(self):
         constraint.validate = mc_widget.validate
 
         # Which row is the constrained parameter in?
+        if model_key == 'poly' and 'Distribution' in constraint.param:
```
Comment (rozyczko) [inline src/sas/qtgui/Perspectives/Fitting/FittingWidget.py:1081]: Shouldn't "Distribution" be enough? All PD parameter names start with "Distribution" and I'm not sure whether any have "poly" in them

File: src/sas/qtgui/Perspectives/Fitting/MultiConstraint.py [hunk 2]
```diff
@@ -102,6 +102,9 @@ def validateFormula(self):
         """
         Add visual cues when formula is incorrect
         """
+        # temporarily disable validation, as not yet fully operational
+        return
```
Comment (rozyczko) [inline src/sas/qtgui/Perspectives/Fitting/MultiConstraint.py:106]: This should be made a separate issue then

**Comment by rozyczko** [comment]
> 1. Revise this PR ignoring the PD constraints for now.
> 2. Revise the PR for [Add All button in Complex Constraint window does not work for polydisperse parameters #2356](https://github.com/SasView/sasview/issues/2356).
> 3. Once that both are accepted and merged with the release branch we can check if add/remove/edit constraints works also as expected for the PD ones.


Or maybe just merge changes from #2356 into this PR to make it a single commit/PR?
Would that be an option?

**Comment by gonzalezma** [comment]
@rozyczko If you think it's preferable to merge changes from #2356 to this PR and delete the other one, that's ok for me. Shall I go ahead with this solution?

**Comment by gonzalezma** [comment]
OK. I just merged the branch corresponding to issue #2356 and everything seems to work. So you should be able now to check:
- Remove or edit a constraint, including the PD ones.
- Adding all constraints at the same time (see #2356)
I will remove the PR that I had for the other issue.

**Comment by wpotrzebowski** [comment]
I may be doing something wrong but I have a problem with setting polydispersity parameters. 
I have 2 fit pages both with polydispersity enabled and at least one polydispersity parameter selected, however I cannot add polydispersity constraints. 
 
(They are not visible from drop-down menu)
Screenshot 2024-01-20 at 07 37 46

(And "Adding all" doesn't work either)
Screenshot 2024-01-20 at 07 38 13

**Comment by gonzalezma** [comment]
That's strange. Have you checked the polydispersity for M2 too and made at least 1 PD param on M2 fittable?

**Comment by wpotrzebowski** [comment]
@gonzalezma ahh, sorry my bad I used push not pull request installer that didn't have changes from the earlier merged PR.

**Review by wpotrzebowski — APPROVED** [review]

**Comment by wpotrzebowski** [comment]
It works fine and in my opinion is ready to be merged.

**Pull request merged**